From 07421276b933b2744c3018f63ecabed50534fd14 Mon Sep 17 00:00:00 2001
From: dianlujitao <dianlujitao@lineageos.org>
Date: Sat, 16 Mar 2019 12:32:40 +0800
Subject: [PATCH 7/7] sepolicy: Allow perf HAL to set mpctl props

 * Addresses the following errors caught in a log:

    [  165.283569] selinux: avc:  denied  { set } for property=vendor.min_freq_0 pid=544 uid=0 gid=0 scontext=u:r:hal_perf_default:s0 tcontext=u:object_r:vendor_mpctl_prop:s0 tclass=property_service permissive=1

Change-Id: I2667b33de063410b93f6805c8fa66ce6ff028eed
---
 vendor/common/hal_perf_default.te | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/vendor/common/hal_perf_default.te b/vendor/common/hal_perf_default.te
index 4083965c..23bd2900 100644
--- a/vendor/common/hal_perf_default.te
+++ b/vendor/common/hal_perf_default.te
@@ -52,6 +52,9 @@ hal_client_domain(hal_perf_default, hal_graphics_composer);
 allow hal_perf_default qdisplay_service:service_manager find;
 r_dir_file(hal_perf_default, appdomain);
 
+# Allow perf HAL to set mpctl props
+set_prop(hal_perf_default, vendor_mpctl_prop)
+
 allow hal_perf {
     sysfs_devices_system_cpu
     sysfs_mpdecision
-- 
2.17.1

